src/lib/engine/facts.js

```diff
@@ -94,33 +95,32 @@ export function addFactsToSpaceGraph({metrics, guesstimates, simulations}, globa
   // When dynamically generating new metrics, we need non-colliding readableIds, so we'll store a running copy of those
   // used, and initialize some variables to account for the extra objects we'll build.
   let readableIds = metrics.map(m => m.readableId)
-  let [factMetrics, factGuesstimates, factSimulations, factHandleMap] = [[], [], [], {}]
+  let [factMetrics, factGuesstimates, factSimulations, factIdMap] = [[], [], [], {}]
   grouped.forEach(([handle, selector, fact]) => {
     // We construct virtual metrics, guesstimates, and simulations from the selector, fact, and running readable IDs...
     const {metric, guesstimate, simulation} = buildFullNode(selector, fact, readableIds)
 
     // Then update all the running variables.
-    factHandleMap[handle] = metric.readableId
+    const idToTranslate = !!_.get(fact, 'id') ? `\$\{fact:${fact.id}\}` : handle
+    factIdMap[idToTranslate] = `\$\{metric:${metric.id}\}`
+
     readableIds = [...readableIds, metric.readableId]
+
     factMetrics = [...factMetrics, metric]
     factGuesstimates = [...factGuesstimates, guesstimate]
     factSimulations = [...factSimulations, simulation]
   })
 
   return {
     metrics: [...metrics, ...factMetrics],
-    guesstimates: [...guesstimates.map(_guesstimate.translateFactHandleFn(factHandleMap)), ...factGuesstimates],
+    guesstimates: [...guesstimates.map(_guesstimate.translateFactHandleFn(factIdMap)), ...factGuesstimates],
     simulations: [...simulations, ...factSimulations],
   }
 }
 
-export function simulateFact(fact) {
-  const fakeSelector = ['fact']
-  const guesstimate = toGuesstimate(fakeSelector, fact)
-  const graph = {metrics: [{id: guesstimate.metric}], guesstimates: [guesstimate]}
-  const metricPropagation = new MetricPropagation(guesstimate.metric, [], 0)
-
-  return metricPropagation.simulate(metricPropagation.remainingSimulations[0], graph).then(
-    ({sample: {values, errors}}) => ({values, errors})
-  )
+export function simulateFact(fact, numSamples=5000) {
```

Comment:
Yeah, it is. I can pull that out into a constant somewhere, if that'd be better. Any suggestions as to where? It was stored as an esoteric internal variable to MetricPropagation before, so doesn't exist anywhere properly atm.